transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/acceptor/LimitsIT.java

```diff
@@ -0,0 +1,162 @@
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.gateway.transport.IoHandlerAdapter;
+import org.kaazing.gateway.transport.nio.internal.NioSocketAcceptor;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class LimitsIT {
+    private static String WS_ECHO_SERVICE_ACCEPT_LIMIT = "ws://localhost:8080/echo";
+    private static String WS_PROXY_SERVICE_ACCEPT_LIMIT = "ws://localhost:8010/proxy";
+    private static int PROXY_SERVICE_CONNECT_PORT_LIMIT = 8123;
+    private static String PROXY_SERVICE_CONNECT_LIMIT = "tcp://localhost:" + PROXY_SERVICE_CONNECT_PORT_LIMIT;
+
+    private static String WS_ECHO_SERVICE_ACCEPT_DEFAULT = "ws://localhost:8081/echo";
+    private static String WS_PROXY_SERVICE_ACCEPT_DEFAULT = "ws://localhost:8011/proxy";
+    private static int PROXY_SERVICE_CONNECT_PORT_DEFAULT = 8124;
+    private static String PROXY_SERVICE_CONNECT_DEFAULT = "tcp://localhost:" + PROXY_SERVICE_CONNECT_PORT_DEFAULT;
+    
+    
+    private static int DEFAULT_WEBSOCKET_MAXIMUM_MESSAGE_SIZE = 128*1024;
+    private static int LIMITED_WEBSOCKET_MAXIMUM_MESSAGE_SIZE_125 = 125;
+    public static final long TEST_LIFETIME = TimeUnit.SECONDS.toSeconds(3);
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/limits");
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create(WS_PROXY_SERVICE_ACCEPT_LIMIT))
+                            .connect(URI.create(PROXY_SERVICE_CONNECT_LIMIT))
+                            .type("proxy")
+                            .acceptOption("ws.maximum.message.size", Integer.toString(LIMITED_WEBSOCKET_MAXIMUM_MESSAGE_SIZE_125))
+                        .done()
+                        .service()
+                            .accept(URI.create(WS_ECHO_SERVICE_ACCEPT_LIMIT))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .acceptOption("ws.maximum.message.size", Integer.toString(LIMITED_WEBSOCKET_MAXIMUM_MESSAGE_SIZE_125))
+                        .done()
+
+                        .service()
+                            .accept(URI.create(WS_PROXY_SERVICE_ACCEPT_DEFAULT))
+                            .connect(URI.create(PROXY_SERVICE_CONNECT_DEFAULT))
+                            .type("proxy")
+                            .acceptOption("ws.maximum.message.size", Integer.toString(DEFAULT_WEBSOCKET_MAXIMUM_MESSAGE_SIZE))
```

Comment:
The purpose of the default tests is to verify the gateway does the check by default ( without being told), so you should remove this line (`acceptOption("ws.maximum.message.size",...`)